### PR TITLE
fix proposal for #733

### LINKED ISSUE #733: permanently in absorption mode
### Describe the problem

I have two different installations.
in both I've recently noticed that it no longer comes into float mode. the voltage of the battery is at max_cell_voltage * cells and the maximum cell drift is 0.002V - 0.007V

what can I do?

I first switched to step mode but would like to continue linearly

thanks for the support

### Driver version

newest on dev

### Venus OS device type

Raspberry Pi

### Venus OS version

3.00

### BMS type

Daly Smart BMS

### Cell count

15

### Connection type

Serial USB adapter to RS485

### Config file

```ini
default config, no changes
```

### Relevant log output

```shell
im not have input
```

### Any other information that may be helpful

_No response_

### PR DESCRIPTION
**Pull request by ogurevich**
allow "Measurement" und "Variation Tolerance" to keep in absorption mode within MAX_VOLTAGE_TIME_SEC. Fixes #733

### CODE REVIEW

**Comment by mr-manuel** [comment]
I reopened it and changed the repo from `master` to `dev`.

**Comment by ogurevich** [comment]
> I reopened it and changed the repo from `master` to `dev`.

thank you !

File: etc/dbus-serialbattery/battery.py [hunk 1]
```diff
@@ -250,7 +250,8 @@ def manage_charge_voltage_linear(self) -> None:
         voltageSum = 0
         penaltySum = 0
         tDiff = 0
-
+        # meassurment and variation tolerance in volts
+        measurementToleranceVariation = 0.0215
```
Comment (mr-manuel) [inline etc/dbus-serialbattery/battery.py:254]: I would not use more than 3 decimals. Is there a reason why you chose to use 4?

File: etc/dbus-serialbattery/battery.py [hunk 2]
```diff
@@ -269,7 +270,10 @@ def manage_charge_voltage_linear(self) -> None:
             if self.max_voltage_start_time is None:
                 # start timer, if max voltage is reached and cells are balanced
                 if (
-                    self.max_battery_voltage - utils.VOLTAGE_DROP <= voltageSum
+                    self.max_battery_voltage
+                    - utils.VOLTAGE_DROP
+                    - measurementToleranceVariation
```
Comment (mr-manuel) [inline etc/dbus-serialbattery/battery.py:275]: I don't think that this should be added here, else the problem is only moved to a lower value.

I my opinion the correct should be e.g. start at 55.2, but reset only, if going below 55.1785 (55.2 - `measurementToleranceVariation`).

File: etc/dbus-serialbattery/battery.py (hunk 1 above)
Comment (ogurevich) [reply etc/dbus-serialbattery/battery.py:254]: not really, some value between 0.02 and 0.03 should be Ok, that is not in principle. For me was value 0.022 most optimal. Please set any value of you choice.   (I'm afraid to suggest that as a parameter)

File: etc/dbus-serialbattery/battery.py (hunk 2 above)
Comment (ogurevich) [reply etc/dbus-serialbattery/battery.py:275]: i agree. the first one ist not necessary. you are right.

File: etc/dbus-serialbattery/battery.py (hunk 1 above)
Comment (ogurevich) [reply etc/dbus-serialbattery/battery.py:254]: The optimal value may be related to CELL_VOLTAGE_DIFF_KEEP_MAX_VOLTAGE_UNTIL and also to specific load fluctuations in the entire system.

Comment (ogurevich) [reply etc/dbus-serialbattery/battery.py:254]: Example with Value 0.03
Screenshot 2023-06-26 at 22 51 04

With Value 0.0215 it takes some more time to meet conditions (better for battery)
Screenshot 2023-06-26 at 22 50 33

**Pull request merged**